### PR TITLE
Switch to non-deprecated action for asset upload on release branch

### PR DESCRIPTION
**Pull request by dibarbet**
Test run here - https://github.com/dibarbet/omnisharp-vscode/actions/runs/5337543533/jobs/9673725833
Test release - https://github.com/dibarbet/omnisharp-vscode/releases/tag/vTest

### CODE REVIEW

**Comment by JoeRobich** [comment]
Could we invoked the new task once and give it all the file names?

File: .github/workflows/release-ci.yml [hunk 1]
```diff
@@ -33,99 +33,63 @@ jobs:
...
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      uses: softprops/action-gh-release@v1
       with:
-        upload_url: ${{ github.event.release.upload_url }}
-        asset_path: ./csharp-${{ env.VERSION }}-darwin-arm64.vsix
-        asset_name: csharp-${{ env.VERSION }}-darwin-arm64.vsix
-        asset_content_type: application/zip
+        token: ${{ secrets.GITHUB_TOKEN }}
+        files: ./csharp-${{ env.VERSION }}-darwin-arm64.vsix
```
Comment (JoeRobich) [inline .github/workflows/release-ci.yml:39]: Maybe we can do it all in one go.

```suggestion
        files: |
          ./csharp-${{ env.VERSION }}-darwin-arm64.vsix
          ./csharp-${{ env.VERSION }}-darwin-x64.vsix
          ./csharp-${{ env.VERSION }}-linux-arm64.vsix
          ./csharp-${{ env.VERSION }}-linux-x64.vsix
          ./csharp-${{ env.VERSION }}-alpine-arm64.vsix
          ./csharp-${{ env.VERSION }}-alpine-x64.vsix
          ./csharp-${{ env.VERSION }}-win32-arm64.vsix
          ./csharp-${{ env.VERSION }}-win32-ia32.vsix
          ./csharp-${{ env.VERSION }}-win32-x64.vsix
```

**Review by JoeRobich — APPROVED** [review]

**Comment by dibarbet** [comment]
Testing 1 step now - https://github.com/dibarbet/omnisharp-vscode/actions/runs/5338486389

File: .github/workflows/release-ci.yml [hunk 2]
```diff
@@ -33,99 +33,16 @@ jobs:
 
     - name: Upload release build (darwin-arm64)
```
Comment (JoeRobich) [inline .github/workflows/release-ci.yml:34]:
```suggestion
    - name: Upload release builds
```

**Pull request merged**